htdocs/install/upgrade2.php

```diff
@@ -5137,3 +5144,62 @@ function migrate_export_import_profiles($mode = 'export')
 		print '<tr class="trforrunsql" style=""><td class="wordbreak" colspan="4">'.$langs->trans("NothingToDo")."</td></tr>\n";
 	}
 }
+
+/**
+ * Migrate Rank into contract  line
+ *
+ * @return  void
+ */
+function migrate_contractdet_rank()
+{
+
+	global $db, $langs;
+
+	$error = 0;
+	$resultstring = '';
+
+	$db->begin();
+	print '<tr class="trforrunsql"><td colspan="4">';
+
+	 $sql = 'SELECT c.rowid as cid ,cd.rowid as cdid,cd.rang FROM '.$db->prefix().'contratdet as cd INNER JOIN '.$db->prefix().'contrat as c ON c.rowid=cd.fk_contrat AND cd.rang=0';
```

Comment:
To avoid phpunit error can you use " to quote the php string (the ' is reserved for the sql string inside the phpstring).
This will remove the phpunit error. Not using this rule makes thing not possible to autodetect sql injection by php unit CodingPhpTest.php